server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -230,6 +224,110 @@ public void testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleShard
         assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, contains(in(shardSizes.getShardIdsWithSizeSmallerOrEqual(usableSpace))));
     }
 
+    public void testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleRestores() {
```

Comment:
`testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleShards` was flaky because of this bug, but only failing once every few hundred iterations. This test fails more reliably for the same reason, although still not all that reliably (after around 20-30 iterations on my laptop). I _could_ make it exercise the exact path that hits the bug every time, but it'd be very specific to this one bug and I'd rather have something a little more general to look out for related bugs too.

Reply:
Yeah I was sort of inclined to keep them both but you're right, we're not really testing anything different in the old test.

Reply:
Great, thanks for updating